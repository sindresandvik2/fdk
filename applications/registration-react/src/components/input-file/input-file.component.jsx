import React from 'react';
import PropTypes from 'prop-types';
import './input-file.scss';

export const InputFile = props => {
  const { className, onChange, children } = props;

  return (
    <React.Fragment>
      <label className={`inputfile ${className}`} htmlFor="file">
        <input
          className="inputfile"
          id="file"
          type="file"
          onChange={e => onChange(e)}
        />
        {children}
      </label>
    </React.Fragment>
  );
};

InputFile.defaultProps = {
  className: null
};

InputFile.propTypes = {
  className: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired
};
