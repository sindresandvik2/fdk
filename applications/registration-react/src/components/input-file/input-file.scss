@import '../../../node_modules/designsystemet/fdk-designsystem-bootstrap4/scss/colors.scss';

.inputfile {
  input {
    width: 0.1px;
    height: 0.1px;
    opacity: 0;
    overflow: hidden;
    position: absolute;
    z-index: -1;
  }
}