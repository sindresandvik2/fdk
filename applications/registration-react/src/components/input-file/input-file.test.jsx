import React from 'react';
import { shallow } from 'enzyme';
import { InputFile } from './input-file.component';

test('should render InputFile success correctly with props', () => {
  const wrapper = shallow(<InputFile type="success">Velg fil</InputFile>);
  expect(wrapper).toMatchSnapshot();
});
