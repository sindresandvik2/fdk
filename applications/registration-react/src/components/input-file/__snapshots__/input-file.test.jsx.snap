// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render InputFile success correctly with props 1`] = `
<React.Fragment>
  <label
    className="inputfile null"
    htmlFor="file"
  >
    <input
      className="inputfile"
      id="file"
      onChange={[Function]}
      type="file"
    />
    Velg fil
  </label>
</React.Fragment>
`;
